modules/network-load-balancer/main.tf

```diff
@@ -0,0 +1,87 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# DEPLOY A NETWORK LOAD BALANCER
+# This module deploys a GCP Regional Network Load Balancer
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+terraform {
+  # This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
+  required_version = ">= 0.12"
+}
+
+# ------------------------------------------------------------------------------
+# CREATE FORWARDING RULE
+# ------------------------------------------------------------------------------
+
+resource "google_compute_forwarding_rule" "default" {
+  provider              = google-beta
+  project               = var.project
+  name                  = var.name
+  target                = google_compute_target_pool.default.self_link
+  load_balancing_scheme = "EXTERNAL"
+  port_range            = var.port_range
+  ip_address            = var.ip_address
+  ip_protocol           = var.protocol
+
+  labels = var.custom_labels
+}
+
+# ------------------------------------------------------------------------------
+# CREATE TARGET POOL
+# ------------------------------------------------------------------------------
+
+variable "instances" {
+  description = "List of self links to instances in the pool. Note that the instances need not exist at the time of target pool creation."
+  type        = list(string)
+  default     = []
+}
+
+resource "google_compute_target_pool" "default" {
+  project          = var.project
+  name             = "${var.name}-tp"
+  region           = var.region
+  session_affinity = var.session_affinity
+
+  instances = var.instances
+
+  health_checks = [
+    google_compute_http_health_check.default.name,
+  ]
+}
+
+# ------------------------------------------------------------------------------
+# CREATE HEALTH CHECK
+# ------------------------------------------------------------------------------
+
+resource "google_compute_http_health_check" "default" {
```

Comment:
Am I correct in assuming that the network load balancer is similar to NLBs in AWS? If so, should we create this health check in the module, since this might be used for non-http endpoints?

Reply:
As per this: https://www.terraform.io/docs/providers/google/r/compute_target_pool.html#health_checks, the http health check is the only supported health check. Though, might make sense to make the health check optional with `var.health_check_enabled`.

Reply:
Makes sense. 👍 to adding a feature flag.

Reply:
Feature flag added